hudi-common/src/main/java/org/apache/hudi/common/fs/TimedSizeAwareOutputStream.java

```diff
@@ -20,19 +20,21 @@
 
 import org.apache.hudi.exception.HoodieException;
 
-import org.apache.hadoop.fs.FSDataOutputStream;
 import org.apache.hadoop.fs.Path;
 
 import java.io.IOException;
+import java.io.OutputStream;
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicLong;
 
 /**
- * Wrapper over <code>FSDataOutputStream</code> to keep track of the size of the written bytes. This gives a cheap way
+ * Wrapper over <code>OutputStream</code> to keep track of the size of the written bytes. This gives a cheap way
  * to check on the underlying file size.
  */
-public class SizeAwareFSDataOutputStream extends FSDataOutputStream {
+public class TimedSizeAwareOutputStream extends OutputStream {
```

Comment:
I see we don't have unit tests for these FileSystems. If incase you plan to write one for newly introduced fileSystem, can you please try if we can add for these too. If not, can you create a jira for testing gaps and tag it here.

Reply:
These file systems are tested by most of the unit tests which read/write data.